### PR TITLE
Add zap logger to tchannel logger adapter

### PR DESCRIPTION
**Pull request by ChuntaoLu**

### CODE REVIEW

File: runtime/tchannel_logger.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	"fmt"
+
+	"github.com/uber/tchannel-go"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// NewTChannelLogger creates a TChannel logger given a zap logger
+func NewTChannelLogger(logger *zap.Logger) tchannel.Logger {
+	return &TChanLogger{logger}
```
Comment (Raynos) [inline runtime/tchannel_logger.go]: Can we return struct by value ? `TChanLogger{logger: logger}`

Also nitpick: always use {logger: logger} for no ambiguity

File: runtime/tchannel_logger.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+// Debug logs a message at debug priority.
+func (l TChanLogger) Debug(msg string) {
+	l.logger.Debug(msg)
+}
+
+// Fields returns the fields that this logger contains.
+func (l TChanLogger) Fields() tchannel.LogFields {
+	// TODO: (lu) is it ok to return nil?
```
Comment (Raynos) [inline runtime/tchannel_logger.go]: I can't find any callers to Fields() in tchannel-go other then unit tests.

Comment (ChuntaoLu) [inline runtime/tchannel_logger.go]: Nice

File: runtime/tchannel_logger.go (hunk 1 above)
Comment (ChuntaoLu) [inline runtime/tchannel_logger.go]: will do

**Pull request merged**